src/process.cpp

```diff
@@ -426,3 +457,8 @@ void remove_timed_out_processes() {
 }
 
 void garbage_collect_processes() { garbage_collect_inodeproc(); }
+
+bool process_exists(pid_t pid) {
+  // No signal is sent! Existance is checked only
+  return kill(pid, 0) != -1;
```

Comment:
I am not sure if 
```c
kill(pid, 0);
```
is posix ...

Reply:
Because, on Linux, I can use the signal zero to check if the process exists ...   
I have found this [documentation](https://www.unix.com/man-page/FreeBSD/2/kill/) about the kill system call. 
It does not mention the signal *0*.

Reply:
Also on linux the docs say:
```
this can be used to check for the existence of a process ID or process group ID that the caller is permitted to signal.
```

In theory I'd say nethogs should be allowed to monitor processes that it's not permitted to signal - though that's probably a rather rare corner case. I'd be OK with removing this check, but it's up to you.